### PR TITLE
Class structure, renaming of Data

### PR DESCRIPTION
**Pull request by MSVoss**
Revision of the reader, introduction of the class structure in the reader of the test suite, renaming of Data.py to config.py(Python standard), introduction of the documentation of used input parameters in the log file

### CODE REVIEW

File: lib/python/test/testsuite/_checkData.py [hunk 1]
```diff
@@ -132,22 +132,22 @@ def checkDirection(variable: str = "undefined",
...
     -------
     out : bool
           False if the variable does not exist or is None,
           else True
     """
-    val_name = "Data." + variable
+    val_name = "config." + variable
 
-    if variable not in dir(Data):
+    if variable not in dir(config):
```
Comment (PrometheusPi) [inline lib/python/test/testsuite/_checkData.py:150]: `config` is a module, what does `dir(config)` thus return?

File: lib/python/test/testsuite/Reader/dataReader.py [hunk 2]
```diff
@@ -31,249 +31,208 @@
     returns the values of the searched parameter
 """
 
-__all__ = ["checkDatFilesInDir",
-           "getAllDatFiles",
-           "allParamsinFile",
-           "getDatwithParam",
-           "getValue"]
+__all__ = ["DataReader"]
```
Comment (PrometheusPi) [inline lib/python/test/testsuite/Reader/dataReader.py:34]: Could you briefly explain, how this reduction is possible?

File: lib/python/test/testsuite/Reader/dataReader.py [hunk 3]
```diff
@@ -31,249 +31,208 @@
...
+        -------
+        ValueError:
+            if the passed parameter is not a .dat file.
+
+        Return:
+        -------
+        out : List of all parameters stored in the file
+        """
+
+        if ".dat" not in file:
```
Comment (PrometheusPi) [inline lib/python/test/testsuite/Reader/dataReader.py:96]: This looks very text files based. How about json etc.?

**Review by PrometheusPi — APPROVED** [review]
Looks good to me, just some minor questions and I will do a run time test.

**Comment by PrometheusPi** [comment]
Run time test was good. 
I just got some warnings:
```
/__path__/pr_4604/picongpu/lib/python/test/setups/ESKHI/../../testsuite/_checkData.py:218: UserWarning: Both config.step_direction and the optional parameter are empty.
  warnings.warn(Both  + val_name +  and the optional parameter
/__path__/pr_4604/picongpu/lib/python/test/setups/ESKHI/../../testsuite/_checkData.py:207: UserWarning: Both config.acceptance and the optional parameter are set. Note that the value from config.py is used.
  warnings.warn(Both  + val_name +  and the optional
/__path__/pr_4604/picongpu/lib/python/test/setups/ESKHI/../../testsuite/_checkData.py:218: UserWarning: Both config.plot_type and the optional parameter are empty.
  warnings.warn(Both  + val_name +  and the optional parameter
/__path__/pr_4604/picongpu/lib/python/test/setups/ESKHI/../../testsuite/_checkData.py:218: UserWarning: Both config.plot_xaxis and the optional parameter are empty.
  warnings.warn(Both  + val_name +  and the optional parameter
/__path__/pr_4604/picongpu/lib/python/test/setups/ESKHI/../../testsuite/_checkData.py:218: UserWarning: Both config.plot_log and the optional parameter are empty.
  warnings.warn(Both  + val_name +  and the optional parameter
/__path__/pr_4604/picongpu/lib/python/test/setups/ESKHI/../../testsuite/_checkData.py:218: UserWarning: Both config.plot_title and the optional parameter are empty.
  warnings.warn(Both  + val_name +  and the optional parameter
/__path__/pr_4604/picongpu/lib/python/test/setups/ESKHI/../../testsuite/_checkData.py:207: UserWarning: Both config.title and the optional parameter are set. Note that the value from config.py is used.
  warnings.warn(Both  + val_name +  and the optional
{'gamma': 1.021, 'BASE_DENSITY_SI': 1e+25, 'DELTA_T_SI': 1.5394e-16}
/__path__/pr_4604/picongpu/lib/python/test/setups/ESKHI/../../testsuite/_checkData.py:101: UserWarning: Both resultDirection and direction are set.Note that the value from config.py is used.
  warnings.warn(Both  + variable +  and direction are set.

```

File: lib/python/test/testsuite/_checkData.py (hunk 1 above)
Comment (MSVoss) [reply lib/python/test/testsuite/_checkData.py:150]: All variables and functions that are in the global namespace of config.py

File: lib/python/test/testsuite/Reader/dataReader.py (hunk 3 above)
Comment (MSVoss) [reply lib/python/test/testsuite/Reader/dataReader.py:96]: For json etc, please use the reader provided

**Pull request merged**